libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/DataStore.java

```diff
@@ -118,16 +118,14 @@ public List<T> getItems(String partitionValue) {
                 .toList();
     }
 
-    public List<T> getItemsWithAttributeLessThanOrEqualValue(
-            String partitionValue, String filterAttribute, String filterValue) {
+    public List<T> getItemsWithBooleanAttribute(String partitionValue, String name, boolean value) {
         var queryConditional =
                 QueryConditional.keyEqualTo(Key.builder().partitionValue(partitionValue).build());
-        AttributeValue expressionValue = AttributeValue.builder().s(filterValue).build();
         var filterExpression =
                 Expression.builder()
-                        .expression("#a <= :b")
-                        .putExpressionName("#a", filterAttribute)
-                        .putExpressionValue(":b", expressionValue)
+                        .expression("#a = :b")
+                        .putExpressionName("#a", name)
+                        .putExpressionValue(":b", AttributeValue.builder().bool(value).build())
```

Comment:
Yep, it works. I did think about doing the filtering client side, but then figured if we've got the ability to do it in the query it might make things a bit cleaner.

Reply:
Ah - you can have both, but this one happens post-query (but before sending to us):
>[FilterExpression](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html#API_Query_RequestSyntax)
A string that contains conditions that DynamoDB applies after the Query operation, but before the data is returned to you. Items that do not satisfy the FilterExpression criteria are not returned.
>
>A FilterExpression does not allow key attributes. You cannot define a filter expression based on a partition key or a sort key.

(I was thinking of the `KeyCondition`, which changes what's actually read from the table)